### PR TITLE
Clean up build

### PR DESCRIPTION
**Pull request by basil**
See self-review for explanation.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,29 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
     <groupId>org.jenkins-ci</groupId>
     <artifactId>jenkins</artifactId>
     <version>1.87</version>
+    <relativePath />
```
Comment (basil) [inline pom.xml:9]: Consistent with the plugin archetype.

File: pom.xml [hunk 2]
```diff
@@ -1,29 +1,31 @@
...
+  <version>${revision}${changelist}</version>
   <name>Pipeline Steps Doc Generator Indexer</name>
   <description>List up extension points and their implementations.</description>
 
   <properties>
-    <gitHubRepo>jenkins-infra/pipeline-steps-doc-generator</gitHubRepo>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <revision>1.0</revision>
+    <changelist>-SNAPSHOT</changelist>
+    <gitHubRepo>jenkins-infra/${project.artifactId}</gitHubRepo>
```
Comment (basil) [inline pom.xml:21]: Consistent with the plugin archetype.

File: pom.xml [hunk 3]
```diff
@@ -1,29 +1,31 @@
...
   <groupId>org.jenkins-ci</groupId>
   <artifactId>pipeline-steps-doc-generator</artifactId>
-  <version>1.0-SNAPSHOT</version>
+  <version>${revision}${changelist}</version>
   <name>Pipeline Steps Doc Generator Indexer</name>
   <description>List up extension points and their implementations.</description>
 
   <properties>
-    <gitHubRepo>jenkins-infra/pipeline-steps-doc-generator</gitHubRepo>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (basil) [inline pom.xml:18]: Present in the parent POM already.

File: pom.xml [hunk 4]
```diff
@@ -1,29 +1,31 @@
...
+    <changelist>-SNAPSHOT</changelist>
+    <gitHubRepo>jenkins-infra/${project.artifactId}</gitHubRepo>
     <spotbugs.excludeFilterFile>${project.basedir}/src/spotbugs/spotbugs-excludes.xml</spotbugs.excludeFilterFile>
   </properties>
 
   <build>
     <plugins>
       <plugin>
         <artifactId>maven-compiler-plugin</artifactId>
-        <version>3.10.1</version>
```
Comment (basil) [inline pom.xml:26]: Present in the parent POM already.

File: pom.xml [hunk 5]
```diff
@@ -41,15 +43,17 @@
             </manifest>
           </archive>
         </configuration>
-        <version>3.2.2</version>
```
Comment (basil) [inline pom.xml:44]: Present in the parent POM already.

File: pom.xml [hunk 6]
```diff
@@ -58,7 +62,6 @@
             </configuration>
           </execution>
         </executions>
-        <version>2.6</version>
```
Comment (basil) [inline pom.xml:61]: Present in the parent POM already at a newer version.

File: pom.xml [hunk 7]
```diff
@@ -41,15 +43,17 @@
...
         <artifactId>maven-assembly-plugin</artifactId>
         <executions>
           <execution>
+            <!-- this is used for inheritance merges -->
+            <id>make-assembly</id>
             <goals>
-              <goal>attached</goal>
+              <goal>single</goal>
             </goals>
+            <!-- bind to the packaging phase -->
```
Comment (basil) [inline pom.xml:56]: Consistent with [the documentation](https://maven.apache.org/plugins/maven-assembly-plugin/usage.html). Needed to adapt to the newer version specified in the parent POM.

**Review by timja — APPROVED** [review]

**Pull request merged**